test/e2e/storage/openstack_stress.go

```diff
@@ -0,0 +1,111 @@
+		scNames       = []string{storageclass1, storageclass2, storageclass3, storageclass4}
+		policyName    string
+		datastoreName string
+	)
+
+	BeforeEach(func() {
+		framework.SkipUnlessProviderIs("openstack")
+		client = f.ClientSet
+		namespace = f.Namespace.Name
+
+		nodeList := framework.GetReadySchedulableNodesOrDie(f.ClientSet)
+		Expect(nodeList.Items).NotTo(BeEmpty(), "Unable to find ready and schedulable Node")
+
+		instancesStr := os.Getenv("OS_STRESS_INSTANCES")
+		Expect(instancesStr).NotTo(BeEmpty(), "ENV OS_STRESS_INSTANCES is not set")
+		instances, err = strconv.Atoi(instancesStr)
+		Expect(err).NotTo(HaveOccurred(), "Error Parsing OS-STRESS-INSTANCES")
+		Expect(instances <= volumesPerNode*len(nodeList.Items)).To(BeTrue(), fmt.Sprintf("Number of Instances should be less or equal: %v", volumesPerNode*len(nodeList.Items)))
+		Expect(instances > len(scNames)).To(BeTrue(), "OS_STRESS_INSTANCES should be greater than 3 to utilize all 4 types of storage classes")
+
+		iterationStr := os.Getenv("OS_STRESS_ITERATIONS")
+		Expect(instancesStr).NotTo(BeEmpty(), "ENV OS_STRESS_ITERATIONS is not set")
+		iterations, err = strconv.Atoi(iterationStr)
+		Expect(err).NotTo(HaveOccurred(), "Error Parsing VCP_STRESS_ITERATIONS")
+		Expect(iterations > 0).To(BeTrue(), "OS_STRESS_ITERATIONS should be greater than 0")
+
+		policyName = os.Getenv("OPENSTACK_SPBM_POLICY_NAME")
+		datastoreName = os.Getenv("OPENSTACK_DATASTORE")
+		Expect(policyName).NotTo(BeEmpty(), "ENV OPENSTACK_SPBM_POLICY_NAME is not set")
+		Expect(datastoreName).NotTo(BeEmpty(), "ENV OPENSTACK_DATASTORE is not set")
+	})
+
+	It("openstack stress tests", func() {
+		scArrays := make([]*storageV1.StorageClass, len(scNames))
+		for index, scname := range scNames {
+			By(fmt.Sprintf("Creating Storage Class : %v", scname))
+			var sc *storageV1.StorageClass
+			var err error
+			switch scname {
+			case storageclass1:
+				sc, err = client.StorageV1().StorageClasses().Create(getOpenstackStorageClassSpec(storageclass1, nil))
+			case storageclass2:
+				var scVSanParameters map[string]string
+				scVSanParameters = make(map[string]string)
+				scVSanParameters[Policy_HostFailuresToTolerate] = "1"
+				sc, err = client.StorageV1().StorageClasses().Create(getOpenstackStorageClassSpec(storageclass2, scVSanParameters))
+			case storageclass3:
+				var scSPBMPolicyParameters map[string]string
+				scSPBMPolicyParameters = make(map[string]string)
+				scSPBMPolicyParameters[SpbmStoragePolicy] = policyName
+				sc, err = client.StorageV1().StorageClasses().Create(getOpenstackStorageClassSpec(storageclass3, scSPBMPolicyParameters))
+			case storageclass4:
+				var scWithDSParameters map[string]string
+				scWithDSParameters = make(map[string]string)
+				scWithDSParameters[Datastore] = datastoreName
+				scWithDatastoreSpec := getOpenstackStorageClassSpec(storageclass4, scWithDSParameters)
+				sc, err = client.StorageV1().StorageClasses().Create(scWithDatastoreSpec)
+			}
```

Comment:
There's better code reuse by removing dup calls to `Create` in this switch a bit, eg:
```
...
scParms := make(map[string]string)
switch scname {
case storageclass1:
      scParms = nil
case storageclass2:
      scParms[Policy_HostFailuresToTolerate] = "1"
case storageclass3:
      scParms[SpbmStoragePolicy] = policyName
case storageclass4:
      scParms[Datastore] =  datastoreName
}
sc, err = client.StorageV1().StorageClasses().Create(getOpenstackStorageClassSpec(storageclass3, scParms))
Expect ...

```